dev_docs/tutorials/adding_performance_metrics.mdx

```diff
@@ -0,0 +1,68 @@
+- `Flexibility` - Teams should be able to flexibly design their event structure without having to define custom mappings (except for edge cases). I.e. a team can decide that the first free field is always used for that page load.
+- `Ease of visualization` - it should be easy to visualize the data
+- `Index field explosion` - having too many field names in the index, damages the performance when visualizing the data.
+
+And event schema is composed of fields which is designed for specific purposes:
+
+- `Event name` We decided to report all metric events under a single event_type, so that we can avoid registering it multiple times. This requires us to add a special standard field for eventName.
+- `Standardized fields` (duration) these fields represent common values that might be reported for performance. This can include duration for time based reports, status, heapSize, memorySize, etc. Standardized fields will be mapped as is to the index, so they can be visualized.
+- `Free fields` (key1,value1,key2, value2, ..., key5, value5) A limited set of numbered free key-value pairs that can be used to report additional use case specific fields. Teams may use them to create their own internal sub-schema. For example, a team may choose to use key1 to always report the loading time of a certain component that is always present on the page. Free fields will be mapped, if present, to the index, so they can be visualized. If a free field becomes popular, it can be promoted to a standardized field, freeing up the free field for other use.
+- `Non standard fields` (kibana_version, protocol ...) - additional fields may be added to an event, by extending them to the meta object. However those won’t be automatically mapped to the index. Since the properties field is a Flattened Field, this means those fields will be searchable by default, but an additional field mapping would be required to visualize them.
+
+Events should be meaningful and can have multiple sub metrics which will give specific information of certain action. For example 
+page-load event can be composed of render time, data load time during the page-load and so on. It's important to understand these 
+events will have meaning for performance investigations and that can be used in visualizations, aggregations. Considering this, 
+creating an event for cpuUsage does not bring any value because it doesn't bring any context with itself and reporting multiple of these 
+events in different places of code will have so much variability during performance analysis of your code. However it can be nice attribute
+to follow if it's important for you to look inside of specific event e.g. page-load. 
+
+# Analytics Client
+
+Holds the public APIs to report events, enrich the events' context and set up the transport mechanisms. Please checkout package documentation to get more information about
+[Analytics Client](https://github.com/elastic/kibana/blob/main/packages/analytics/README.md).
```

Comment:
Is this a left-over?

Reply:
It's mentioned in the [issue](https://github.com/elastic/kibana/issues/136726) to have 
> Introduction to the AnalyticsClient

so I believe we want to briefly mention about Analytics Client

Reply:
I'm resolving this and keeping it for a while since we didn't get any comment on this. @afharo please re-open if you think it's important not to have it in here.